### DISCUSSION TITLE
What is the difference between ShippingMethod and ShippingMethodTypes

### DISCUSSION THREAD

**Discussion by eric-makro** [saleor/saleor#10610] [Q&A]
Can someone help please what is the difference between ShippingMethod and ShippingMethodTypes?
I see on database table there is only shipping_shippingmethod. May I know how can we differentiate these?
Thanks

**Comment by maarcingebala** [comment]
Hi @eric-makro, they actually represent the same object in DB, but the difference is that `ShippingMethod` is used in "public" API, so all storefront/checkout related queries and mutations; while `ShippingMethodType` is used in all admin-privileged operations. `ShippingMethodType` contains some additional configuration fields which admins can use, but they are not visible to the shop customers.

**Reply by korycins** [reply]
Hi @eric-makro 
To extend what Marcin wrote:
If we're talking from API perspective, it is not the same. ShippingMethod is a type that is directly assigned to Checkout/Order. It can be also an [external shipping method fetched from the App](https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/shipping-webhooks#list-external-shipping-methods) or built-in shipping method. ShippingMethodType is a type that is used to configure the built-in shipping methods. Like providing pricing for each channel etc. Based on, provided details to ShippingMethodType, Saleor returns a list of built-in ShippingMethods and includes the external ones. The ShippingMethod type doesn't have any fields related to configurations. So for example: developer that builds the storefront doesn't need to think about which price field should be used.
